data_collection/web/javascript_tracking_client/api.rst

```diff
@@ -0,0 +1,1800 @@
+or :ref:`Piwik.getAsyncTracker<jtc-api-Piwik.getAsyncTracker>` function.
+
+.. _jtc-api-Piwik.getTracker:
+
+.. function:: Piwik.getTracker(trackerUrl, siteId)
+
+    Getter for Tracker object.
+
+    :param string trackerUrl: **Required** URL for Tracker
+    :param string siteId: **Required** Site ID that will be linked to tracked data.
+    :returns: Tracker object
+
+    Example of usage (accessing Tracker object and tracking a custom event)::
+
+        var tracker = Piwik.getTracker("https://example.com/", "45e07cbf-c8b3-42f3-a6d6-a5a176f623ef");
+        tracker.trackEvent("video", "video-paused", "intro.mp4", 15.2);
+
+.. _jtc-api-Piwik.getAsyncTracker:
+
+To access internal Tracker object used for asynchronous tracking you must use
+the ``Piwik.getAsyncTracker``.
+
+.. function:: Piwik.getAsyncTracker(trackerUrl, siteId)
+
+    Getter for Tracker instance.
+
+    :param string trackerUrl: **Required** URL for Tracker
+    :param string siteId: **Required** Site Id that will be linked to tracked data.
+    :returns: Tracker instance
+
+    Example of usage (accessing Tracker object and tracking a custom event)::
+
+        var tracker = Piwik.getAsyncTracker("https://example.com/", "45e07cbf-c8b3-42f3-a6d6-a5a176f623ef");
+        tracker.trackEvent("video", "video-paused", "intro.mp4", 15.2);
+
+    Tracker object is also accessible through ``this`` keyword in a special
+    command pushed to command queue, where the first element of the command
+    array is a custom function.::
+
+        _paq.push([function () {
+            // *this* is a Tracker object
+            this.addEcommerceItem("01725334", "USB-C chord")
+            console.log(this.getEcommerceItems());
+        }]);
+
+    .. warning::
+
+        Tracker object can't be accessed before JavaScript Tracking Client
+        loads.
+
+.. _jtc-api-tracking-functions:
+
+Tracking functions
+------------------
+
+Tracking functions collect and send data to tracking backend. They can by
+called on a :ref:`tracker object<jtc-api-tracker-object>` or pushed to
+the :ref:`command queue<jtc-api-command-queue>` as commands.
+
+
+
+
+
+.. _jtc-api-page-views:
+
+Page views
+^^^^^^^^^^
+
+.. _jtc-api-trackPageView:
+
+.. function:: trackPageView([customPageTitle])
+
+    Tracks page view of the page that the function was run on.
+
+    :param string customPageTitle: **Optional** Custom page title, used only for this event
+
+    Example of usage::
+
+        [command queue]
+        _paq.push(["trackPageView"]);
+        [tracker object]
+        tracker.trackPageView();
+
+    .. note::
+
+        To overwrite page title for **all events** that will happen on the page
+        (until a reload), use :ref:`setDocumentTitle<jtc-api-setDocumentTitle>`
+        function.
+
+    .. note::
+
+        ``trackPageView`` is included in the default tracker setup snippet.
+        It's likely you're already using it.
+
```

Comment:
Why so many new lines?

Reply:
It's really hard to navigate a big rst document. I've added 5 newlines between sections to make it clear where a new sections starts as you scroll the file.